sig/unit_test/with_aliases.rbs

```diff
@@ -0,0 +1,136 @@
+module RBS
+  module UnitTest
+    # WithAliases module provides `with_***` methods that helps writing tests which passes convertible objects
+    #
+    # ```ruby
+    # include WithAliases
+    #
+    # with_int do |i|
+    #   # Yields Integer and RBS::UnitTest::Convertibles::ToInt
+    # end
+    #
+    # with_string.and_nil do |s|
+    #   # Yields String, RBS::UnitTest::Convertibles::ToStr, and nil
+    # end
+    # ```
+    #
+    module WithAliases
+      include Convertibles
+
+      class WithEnum[T]
+        @enum: _Each[T]
+
+        include Enumerable[T]
+
+        def initialize: (_Each[T] enum) -> void
+
+        def each: () { (T) -> void } -> void
+
+        # Yield `nil` in addition to original `enum` elements
+        #
+        # ```ruby
+        # e = WithEnum.new([1, 2])
+        # e.and_nil do |i|
+        #   pp i     # => 1, 2, and nil
+        # end
+        # ```
+        #
+        def and_nil: () { (T?) -> void } -> void
+                   | () -> WithEnum[T?]
+
+        # Skip yielding objects in `cases`
+        #
+        # ```ruby
+        # e = WithEnum.new([1, 2, 3])
+        # e.but(2) do |i|
+        #   pp i     # => 1 and 3
+        # end
+        # ```
+        #
+        def but: (*T cases) { (T) -> void } -> void
+               | (*T cases) -> WithEnum[T]
+
+        # Yield objects in `args` in addition to original `enum` elements
+        #
+        # ```ruby
+        # e = WithEnum.new([1, 2])
+        # e.and(3) do |i|
+        #   pp i     # => 1, 2, and 3
+        # end
+        # ```
+        #
+        def and: [S] (*S args) { (T | S) -> void } -> void
+               | [S] (*S args) -> WithEnum[T | S]
+      end
+
+      # Yields `args` objects
+      #
+      def with: (*untyped args) -> void
+              | (*untyped args) { (untyped) -> void } -> void
+
+      # Yields `::int` objects
+      #
+      def with_int: (?Integer value) { (int) -> void } -> void
+                  | (?Integer value) -> WithEnum[int]
+
+      # Yields `::float` objects
+      #
+      def with_float: (?Float value) { (float) -> void } -> void
+                    | (?Float value) -> WithEnum[float]
+
+      # Yields `::string` objects
+      #
+      def with_string: (?String value) { (string) -> void } -> void
+                     | (?String value) -> WithEnum[string]
+
+      # Yields `::array` objects
+      #
+      def with_array: (*untyped elements) { (array[untyped]) -> void } -> void
```

Comment:
We should give this a type, like `def with_array: [T] (*T elements)`. Same with `hash` and `range`

Reply:
Skip `range` method because it requires `<=>` method.